### PR TITLE
Add errors for Only H1 headings

### LINKED ISSUE #5: Initial validation
We should validate the input notebook:

- [ ] if there are no markdown cells, show an error
- [ ] if there are only H1 headings

### PR DESCRIPTION
**Pull request by 94rain**
## Describe your changes
* Add errors for Only H1 headings:
  * Logic:
      * No H2 headers:
          * Only H1 headers: `Only H1 headings are found. At this time, only H2 heading are supported.`
          * No H1 and H2 headers: `Expected notebook to have at least one markdown H2 heading.`
      * Only one H2 header: Warning: `refactoring successful but only one H2 heading detected...`
      * Otherwise: run smoothly
* Add a new test case for `_get_h2_header` (`###`->None), and a new method `_get_h1_header` along with tests
* Update test `test_find_breaks_error_if_no_h2_headers` to `test_find_breaks_error_if_no_h2_but_h1_headers` and `test_find_breaks_error_if_no_h1_and_h2_headers`

## Issue ticket number and link
Closes #5 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.
 - Update: Fixed the issue of matching H3 or smaller headers as H2. Add error messages when there are only H1 headers.

### CODE REVIEW

File: src/soorgeon/split.py [hunk 1]
```diff
@@ -87,7 +96,23 @@ def _get_h2_header(md):
...
+
+        if match:
+            found = _sanitize_name(match.group(1))
+
+            break
+
+    return found
+
+
+def _get_h1_header(md):
```
Comment (edublancas) [inline src/soorgeon/split.py:109]: given that `_get_h1_header` and `_get_h2_header`, we can create a "factory" that generates both, something like this:

```
def _get_header_factory(regex):
    def _get_header(md):
         # put the body of the function here, and pass regex to re.search
    
    return _get_header


_get_h1_header  = _get_header_factory(regex_for_h1)
_get_h2_header  = _get_header_factory(regex_for_h2)
```

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by edublancas** [comment]
looks good, only a small change

**Comment by 94rain** [comment]
> looks good, only a small change

Done. Please take another look.

**Comment by idomic** [comment]
lgtm, nice job!
